### PR TITLE
ci: Update docker images to use Node v14

### PR DESCRIPTION
**Pull request by ActuallyACat**
# Objective
Updating Node to v14

# Motivation and Context 
v14.15.1 is the version Node recommends for most users. 

# Checklist

~~[ ] If this contains visual changes, has it been reviewed by a designer?~~ 
[ ] I have considered likely risks of these changes and got someone else to QA as appropriate
[ ] I have or will communicate these changes to the front end practice

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v12.16.3
+v14.5.0
```
Comment (makenosound) [inline .nvmrc:1]: Should this match the specific node version you’re pinning on the Docker images?

**Review by mbylstra — APPROVED** [review]
I'm guessing this is for Stencil :)

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "site"
   ],
   "engines": {
-    "node": "^12.16.3"
+    "node": "^14.5.0"
```
Comment (makenosound) [inline package.json:10]: Feels like we should perhaps bump this too (even though it’s not strictly necessary). Having it consistent with the rest of the values will mean it’s better kept in sync.

File: .nvmrc (hunk 1 above)
Comment (ActuallyACat) [reply .nvmrc:1]: I _don't think_ a patch will cause us grief 🤔 I will update just to be sure though. Thanks :)

File: package.json (hunk 2 above)
Comment (ActuallyACat) [reply package.json:10]: I'm not sure why, but it appears I can't download v14.5.1 from Node (https://nodejs.org/dist/) using asdf :/

Comment (makenosound) [reply package.json:10]: Oh, `14.5.1` doesn’t exist I think? You want `14.15.1` (which because strings shows up further down the list).

Comment (ActuallyACat) [reply package.json:10]: 😬 I can't read. 14.1.x =/= 14.15.x

**Pull request merged**